components/Common/Sidebar/index.tsx

```diff
@@ -0,0 +1,44 @@
+import type { Url } from 'url';
+
+import { useState } from 'react';
+import type { FC } from 'react';
+
+import styles from './index.module.css';
+import SidebarGroup from './SidebarGroup';
+
+export type SidebarItemType = {
+  url: string | Url;
+  title: string;
+};
+
+export type SidebarGroupType = {
+  groupName: string;
+  items: SidebarItemType[];
+};
+
+export type ActiveItem = Pick<SidebarGroupType, 'groupName'> &
+  Pick<SidebarItemType, 'title'>;
+
+type SidebarProps = {
+  groups: SidebarGroupType[];
+};
```

Comment:
```suggestion
type SidebarProps = {
  groups: Omit<ComponentProps<typeof SidebarGroup>, 'activeItem'>[];
  activeItem?: ComponentProps<typeof SidebarItem>;
};
```

You can define one type for each component and use it with `ComponentProps`. IMO, this more simpler and understandable